src/globals.c

```diff
@@ -212,13 +213,37 @@ bool set_format_string(char *userinput)
 	return true;
 }
 
+bool set_state_dir(char *dir)
```

Comment:
In init_globals(), I would like to call set_state_dir(NULL), but that's not possible in the current state, because dir==NULL will return false. I'm suggesting to make NULL a special value, just like it is for the other set_* functions.